source/class.cpp

```diff
@@ -1172,12 +1177,16 @@ void ClassBinder::bind(Context &context)
 	string const trampoline_name = callback_structure_constructible ? callback_structure_name(C) : "";
 	string const binding_qualified_name = callback_structure_constructible ? callback_structure_name(C) : qualified_name;
 
-	string maybe_holder_type = ", std::shared_ptr<{}>"_format(qualified_name); // for now enable std::shared_ptr by default
-	if( is_inherited_from_enable_shared_from_this(C) ) maybe_holder_type = ", std::shared_ptr<{}>"_format(qualified_name);
+	string holder_type = Config::get().holder_type();
+
+	string maybe_holder_type = ", {}<{}>"_format(holder_type, qualified_name);
```

Comment:
Optional: looks like original code have duplication, this could be refactored as `string maybe_holder_type;`